.github/workflows/check.yaml

```diff
@@ -60,96 +81,69 @@ jobs:
       with:
         include-hidden-files: true
         if-no-files-found: ignore
-        name: coverage-unit-${{ env.SYSTEM_ARCH }}
+        name: coverage-unit-${{ matrix.python-version }}-${{ env.SYSTEM_ARCH }}
         path: .coverage-unit
 
-
-  mocked-plans:
-    name: Mocked plans
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v4
-    - name: Set up Python 3.10
-      uses: actions/setup-python@v5
-      with:
-        python-version: '3.10'
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        python -m pip install tox
-    - name: Run mocked-plans tests
-      run: tox -e mocked-plans
-
-  snap-build:
-    name: Build snap
+  build:
     needs:
       - lint
-      - unit
-      - mocked-plans
-    runs-on: ubuntu-latest
-    timeout-minutes: 30
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [[ubuntu-22.04]]
     steps:
       - uses: actions/checkout@v4
         with:
-          submodules: true
           fetch-depth: 0  # Complete git history is required to generate the version from git tags.
-      - name: Setup Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: '3.10'
-      - name: Install tox
-        run: |
-          python -m pip install --upgrade pip
-          python -m pip install tox
-      - name: Setup LXD
-        uses: canonical/setup-lxd@v0.1.1
-        with:
-          channel: latest/stable
-      - name: Install snapcraft
-        run: sudo snap install snapcraft --classic
-      - name: Build snap
-        run: snapcraft --use-lxd
+
+      - name: Verify snap builds successfully
+        id: build
+        uses: canonical/action-build@v1
+
       - name: Determine system architecture
         run: echo "SYSTEM_ARCH=$(uname -m)" >> $GITHUB_ENV
-      - name: Upload the built snap as an artifact
+
+      - name: Upload the built snap
         uses: actions/upload-artifact@v4
         with:
           name: snap_${{ env.SYSTEM_ARCH }}
-          path: charmed-openstack-upgrader_*.snap
+          path: ${{ steps.build.outputs.snap }}
 
   func:
-    name: Functional tests
-    needs: snap-build
-    runs-on: ubuntu-latest
-    timeout-minutes: 120
+    needs:
+      - build
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [[ubuntu-22.04]]
     steps:
       - uses: actions/checkout@v4
         with:
-          submodules: true
-      - name: Setup Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: '3.10'
-      - name: Setup Juju 3.6/stable environment
-        uses: charmed-kubernetes/actions-operator@main
-        with:
-          provider: lxd
-          juju-channel: 3.6/stable
```

Comment:
Tests are failing because the snap_check template does not include setting up juju

Reply:
Already fixed, closing